runtime/common/nativelocation.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const NativeLocationPrefix = "N"
+
+// NativeLocation
+//
+type NativeLocation struct{}
```

Comment:
I don't think we have to introduce a new location. The other "native" / built-in types don't have a location, e.g. `AuthAccount`'s type ID is just `AuthAccount

Reply:
Yeah, I agree. Had to introduce this to make the exporting type/value work. Currently, the built-in types/values also get exported like any other struct, so that we can give the same user-experience when exporting/importing (also would be consistent in the JSON-Cadence data format as well). 

`cadence.StructType` relies on location. But we can go with a nil location, if that's ok. Thought it would be a bit dangerous for the downstream code to return nil..

Reply:
just realized location is also needed for composite value encoding/decoding to/from CBOR. 
(I have missed adding that part here - one more reason to unify the two composite types :smile: ).

Perhaps would be good to have chat on what we can do here?

Reply:
Removed in https://github.com/onflow/cadence/pull/633/commits/a5abb9669d4770a3a99e3d637581abb6567eae9f.